packages/synapse-sdk/src/pdp/server.ts

```diff
@@ -494,13 +484,13 @@ export class PDPServer {
    * @param data - The data to upload
    * @returns Upload response with PieceCID and size
    */
-  async uploadPiece(data: Uint8Array | ArrayBuffer): Promise<UploadResponse> {
+  async uploadPiece(data: Uint8Array, pieceCid: PieceCID): Promise<void> {
     // Convert ArrayBuffer to Uint8Array if needed
-    const uint8Data = data instanceof ArrayBuffer ? new Uint8Array(data) : data
```

Comment:
I removed the ability of PDPServer.uploadPiece to accept ArrayBuffer but if we actually need that I can put it back.